sdk/component/store/remote/findings-client/client.go

```diff
@@ -168,68 +182,96 @@ func (c *client) Validate(finding *ocsf.VulnerabilityFinding) error {
 
 // Read gets findings by instanceID.
 func (c *client) Read(ctx context.Context, instanceID uuid.UUID) ([]*vf.VulnerabilityFinding, error) {
-	resp, err := c.findingsSvcClient.GetFindings(ctx, &v1.GetFindingsRequest{Id: instanceID.String()})
-	if err != nil {
-		return nil, errors.Errorf("could not get findings: %w", c.checkErr(err))
-	}
-
-	switch {
-	case resp == nil:
-		return nil, errors.New("unexpected nil response")
-	case len(resp.Findings) == 0:
-		return nil, store.ErrNoFindingsFound
-	}
-
-	findings := make([]*vf.VulnerabilityFinding, 0, len(resp.Findings))
-	for _, finding := range resp.Findings {
-		findings = append(
-			findings,
-			&vf.VulnerabilityFinding{
-				Finding: finding.GetDetails(),
-				ID:      finding.GetId(),
+	findingBatches := [][]*v1.Finding{}
+	for page := uint32(0); ; page++ {
+		resp, err := c.findingsSvcClient.GetFindings(
+			ctx,
+			&v1.GetFindingsRequest{
+				Id:       instanceID.String(),
+				Page:     &page,
+				PageSize: &c.pageSize,
 			},
 		)
+		if err != nil {
+			return nil, errors.Errorf("could not get findings: %w", c.checkErr(err))
+		}
+
+		switch {
+		case resp == nil:
+			return nil, errors.New("unexpected nil response")
+		case len(resp.Findings) == 0 && page == 0:
+			return nil, store.ErrNoFindingsFound
+		case len(resp.Findings) < int(c.pageSize):
+			findingBatches = append(findingBatches, resp.Findings)
+			return slices.Collect(
+				utils.MapSlice(
+					utils.CollectSlices(findingBatches),
+					func(finding *v1.Finding) *vf.VulnerabilityFinding {
+						return &vf.VulnerabilityFinding{
+							Finding: finding.GetDetails(),
+							ID:      finding.GetId(),
+						}
+					},
+				),
+			), nil
+		default:
+			findingBatches = append(findingBatches, resp.Findings)
+		}
 	}
-
-	return findings, nil
 }
 
 // Update updates findings by instanceID.
 func (c *client) Update(ctx context.Context, instanceID uuid.UUID, findings []*vf.VulnerabilityFinding) error {
-	reqFindings := make([]*v1.Finding, 0, len(findings))
-	for _, finding := range findings {
-		reqFindings = append(
-			reqFindings,
-			&v1.Finding{
-				Id:      finding.ID,
-				Details: finding.Finding,
-			},
-		)
-	}
-
-	if _, err := c.findingsSvcClient.UpdateFindings(
-		ctx,
-		&v1.UpdateFindingsRequest{
-			Id:       instanceID.String(),
-			Findings: reqFindings,
+	logger := sdklogger.LoggerFromContext(ctx)
+
+	batchNo := 1
+	findingPageIterator := utils.MapSlice(
+		slices.Chunk(findings, int(c.pageSize)),
+		func(page []*vf.VulnerabilityFinding) []*v1.Finding {
+			vf := []*v1.Finding{}
+			for _, finding := range page {
+				vf = append(vf, &v1.Finding{
+					Id:      finding.ID,
+					Details: finding.Finding,
+				})
+			}
+			return vf
 		},
-	); err != nil {
-		return errors.Errorf("could not update findings: %w", c.checkErr(err))
+	)
+	for findingPage := range findingPageIterator {
+		logger.Info("submitting batch of findings", "batchNo", batchNo, "batchSize", len(findingPage))
```

Comment:
⛏️ can we use `slog.Int("batch_no", batchNo)` etc here?

Reply:
fixed